src/pages/InputDemo/InputDemo.jsx

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-console */
+import React from 'react';
+import { TextField, SelectField, RadioField } from '../../components';
+import { selectOptions, radioOptionsCricket, radioOptionsFootball } from '../../config/constants';
+
+class InputDemo extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      sport: '',
+      cricket: '',
+      football: '',
+    };
+  }
+
+    handleNameChange = (e) => {
+      this.setState({ name: e.target.value }, () => {
+        console.log(this.state);
+      });
+    }
+
+    handleSportChange = (e) => {
+      this.setState({ sport: e.target.value }, () => console.log(this.state));
+      if (e.target.value === 'Select') {
+        this.setState({ sport: '' });
+      }
+      return e.target.value === 'cricket' ? this.setState({ football: '' }) : this.setState({ cricket: '' });
+    }
```

Comment:
one setState is enough is make changes change in state. Unnecessary use of setState

Reply:
changed to one state.